docs/user/security/reporting.asciidoc

```diff
@@ -42,57 +55,66 @@ For more information, see {ref}/security-privileges.html[Security privileges].
-`reporting_user` role and your new custom role, `custom_reporting_user`.
-
-[float]
-==== With a custom index
-
-If you are using Reporting with a custom index,
-the `xpack.reporting.index` setting should begin 
-with `.reporting-*`. The default {kib} system user has
-`all` privileges against the `.reporting-*` pattern of indices.
-
-[source,js]
-xpack.reporting.index: '.reporting-custom-index'
-
-If you use a different pattern for the `xpack.reporting.index` setting,
-you must create a custom role with appropriate access to the index, similar
-to the following:
-
-. Open the main menu, then click *Stack Management > Roles*.
-. Click *Create role*, then name the role `custom-reporting-user`.
-. Specify the custom index and assign it the `all` index privilege.
-. Open the main menu, then click *Stack Management > Users* and create a new user with
-the `kibana_system` role and the `custom-reporting-user` role.
-. Configure {kib} to use the new account:
-[source,js]
-elasticsearch.username: 'custom_kibana_system'
+. Open the main menu, then click *Stack Management > Users*, add a new user, and assign the user 
+your new `custom_reporting_user` role.
 
 [float]
 [[reporting-roles-user-api]]
 ==== With the user API
-This example uses the {ref}/security-api-put-user.html[user API] to create a user who has the
-`reporting_user` role and the `kibana_admin` role:
+This example uses the {ref}/security-api-put-role.html[role API] to create a role that
+grants the privilege to generate reports in *Canvas*, *Discover*, *Visualize Library*, and *Dashboard*.
+This role is meant to be granted to users in combination with other roles that grant read access
+to the data in {es}, and at least read access in the applications
+where they'll generate reports.
 
 [source, sh]
 ---------------------------------------------------------------
-POST /_security/user/reporter
+POST /_security/role/custom_reporting_user
 {
-  "password" : "x-pack-test-password",
-  "roles" : ["kibana_admin", "reporting_user"],
-  "full_name" : "Reporting User"
+  metadata: {},
+  elasticsearch: { cluster: [], indices: [], run_as: [] },
+  kibana: [
+    {
+      base: [],
+      feature: {
+        dashboard: [
+          'generate_report', <1>
+          'download_csv_report' <2>
+        ],
+        discover: ['generate_report'], <3>
+        canvas: ['generate_report'], <4>
+        visualize: ['generate_report'], <5>
+      },
+      spaces: ['*'],
+    }
+  ]
 }
 ---------------------------------------------------------------
+// CONSOLE
+
+<1> Grants access to generate PNG and PDF reports in *Dashboard*.
+<2> Grants access to download CSV files from saved search panels in *Dashboard*.
+<3> Grants access to generate CSV reports from saved searches in *Discover*.
+<4> Grants access to generate PDF reports in *Canvas*.
+<5> Grants access to generate PNG and PDF reports in *Visualize Library*.
 
 [float]
 === If you are using an external identity provider
```

Comment:
Change this title as you changed the previous title that started with "If you are using..."  Would "With an external provider" work?

Reply:
How about "When using an external provider"?

This is a higher-level heading than the title that you mentioned.